pkg/scheduler/nodeinfo/node_info.go

```diff
@@ -665,12 +665,14 @@ func (n *NodeInfo) FilterOutPods(pods []*v1.Pod) []*v1.Pod {
 			continue
 		}
 		// If pod is on the given node, add it to 'filtered' only if it is present in nodeInfo.
-		podKey, _ := GetPodKey(p)
-		for _, np := range n.Pods() {
-			npodkey, _ := GetPodKey(np)
-			if npodkey == podKey {
-				filtered = append(filtered, p)
-				break
+		if podKey, err := GetPodKey(p); err == nil {
+			for _, np := range n.Pods() {
```

Comment:
Rather than nesting the "happy path" inside the if statement, it is preferred to continue if `err != nil`